### DISCUSSION TITLE
Firebase 9, Quasar and Composition API

### DISCUSSION THREAD

**Discussion by joshuavachon25** [quasarframework/quasar#11449] [General - Components / Directives / etc] [answered]
Hi!

I'm new to Quasar and to the version of Firebase and I'm not sure to know how it work. I tried to initiate it with a boot file (boot/firebase.js) and then import the db in my index.vue to get data and show it in my template, but I get strange results...

How to use the new Firebase (firestore) with Quasar and the new composition API?

What I did:

**Firebase.js**
```javascript
import { initializeApp } from "firebase/app"
import { getFirestore } from "firebase/firestore";

const firebaseConfig = {
...
};

const app = initializeApp(firebaseConfig);
const db = getFirestore(app)

export default { app, db }
```
**Index.vue**
```html
<template>
 ...
<p>{{articles}}</p>
...
</template>
```
```javascript
import db from "boot/firebase"

export default{
  setup(){
    const articles = async () => {
        return await db.collection('articles').get()
      }
    return{ articles }
  },
}
```
**Result (Inspect element)**
```html
<p>async () =&gt; {
      return await boot_firebase__WEBPACK_IMPORTED_MODULE_2__["default"].collection('articles').get();
    }</p>
```

**Comment by MatthewSH** [comment, marked as answer]
Looks like you're trying to use a method as a computed property, that's why you're getting that result in HTML. You'd need to call the method, `articles()`.
